File: 004/004.brute.int.js
// brute-force solution
// palindrome check via arithmetic integer reverse

const reversed = (number) => {
  /**
   * Reverses an integer.
   * @param {int} number - an integer to reverse.
   * @returns {int} - a reversed integer.
   */
  let inversed = 0;
  while (number) 
  {
    inversed = 10 * inversed + number % 10;
    number = Math.floor(number / 10);
  }
  return inversed;
};

function largestPalindromeProduct(a, b) {
  /**
   * Searches for the largest palindrome product from two integers.
   * @param {int} a - the number of digits in the first number.
   * @param {int} b - the number of digits in the second number.
   * @returns {int} - the largest found palindrome product.
   */
  let largest = 0;
  for (let num1 = 10 ** (a - 1); num1 < 10 ** a; num1++) {
    for (let num2 = 10 ** (b - 1); num2 < 10 ** b; num2++) {
      let product = num1 * num2;
      if (product == reversed(product)) {
        largest = Math.max(largest, product);
      }
    }
  }
  return largest;
}

// results
console.log(largestPalindromeProduct(2, 2));
console.log(largestPalindromeProduct(3, 3));

File: deprecated/004/python/004.brute.int.py
# brute-force solution
# palindrome check by using arithmetics 

def reverse(number: int) -> int:
  """ Reverses an integer. \n
    Args: \n
      number (int): integer to reverse. \n
    Returns: \n
      A reversed integer. \n
  """
  inversed = 0
  while number:
    inversed = 10 * inversed + number % 10
    number //= 10
  return inversed

# palindrome integer check
is_palindrome = lambda number: number == reverse(number)

# search
def largest_palindrome_product(a: int, b: int) -> int:
  largest = 0
  for num1 in range(10 ** (a - 1), 10 ** a):
    for num2 in range(10 ** (b - 1), 10 ** b):
      product = num1 * num2
      if is_palindrome(product):
        largest = max(largest, product)
  return largest

# results
print(largest_palindrome_product(2, 2))
print(largest_palindrome_product(3, 3))